app/src/app/layout/table/table.component.ts

```diff
@@ -103,8 +103,11 @@ export class TableComponent<T> implements OnInit, OnDestroy {
   @Input('columns')
   public set Columns(columns: Array<Column>) {
     this.columns = columns;
-    this.initialColumns = Object.assign([], columns);
-    this.items = [];
+    const temp = [];
+    for (const col of columns) {
+      temp.push(Object.assign({}, col));
+    }
+    this.initialColumns = temp;
```

Comment:
You also took out the reset item array. So wrong values are displayed at new Column set.